core/src/main/java/com/linecorp/armeria/common/metric/MoreMeters.java

```diff
@@ -0,0 +1,166 @@
+        // According to Jon Schneider of Micrometer (discussion at micrometer-metrics.slack.com):
+        // (1) Frugal2U is by far the fastest, but can take a while to converge.
+        // (2) CKMS is slower but isn’t a successive approximation approach.
+        // (3) Window Sketch is the slowest, but the quantile is sensitive to recent samples.
+        return new WindowSketchQuantiles.Builder().quantiles(DEFAULT_QUANTILES).create();
+    }
+
+    /**
+     * Returns a newly-created immutable {@link Map} which contains all values of {@link Meter}s in the
+     * specified {@link MeterRegistry}. The format of the key string is:
+     * <ul>
+     *   <li>{@code <name>#<statistic>{tagName=tagValue,...}}</li>
+     *   <li>e.g. {@code "armeria.server.activeRequests#value{method=greet}"}</li>
+     *   <li>e.g. {@code "someSubsystem.someValue#sumOfSquares"} (no tags)</li>
+     * </ul>
+     * Note: It is not recommended to use this method for the purposes other than testing.
+     */
+    public static Map<String, Double> measureAll(MeterRegistry registry) {
+        checkArgument(registry instanceof AbstractMeterRegistry,
+                      "registry: %s (expected: %s)", registry, AbstractMeterRegistry.class);
+
+        final ImmutableMap.Builder<String, Double> builder = ImmutableMap.builder();
+        getMeterMap(registry).forEach((id, meter) -> stream(meter.measure()).forEach(measurement -> {
+            final String fullName = measurementName(id, measurement);
+            final double value = measurement.getValue();
+            builder.put(fullName, value);
+        }));
+
+        return builder.build();
+    }
+
+    private static String measurementName(Object id, Measurement measurement) {
+        final StringBuilder buf = new StringBuilder();
+
+        // Append name.
+        buf.append(getName(id));
+
+        // Append statistic.
+        buf.append('#');
+        buf.append(CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_CAMEL, measurement.getStatistic().name()));
+
+        // Append tags if there are any.
+        final Iterator<Tag> tagsIterator = getTags(id).iterator();
+        if (tagsIterator.hasNext()) {
+            buf.append('{');
+            tagsIterator.forEachRemaining(tag -> buf.append(tag.getKey()).append('=')
+                                                    .append(tag.getValue()).append(','));
+            buf.setCharAt(buf.length() - 1, '}');
+        }
+        return buf.toString();
+    }
+
+    /**
+     * Micrometer, as of 0.10.0, has a bug where Meter.getName() sometimes returns conventional name
+     * and sometimes not. We use reflection to get the consistent meter names as a workaround.
+     */
+    @SuppressWarnings("unchecked")
+    private static Map<Object, Meter> getMeterMap(MeterRegistry registry) {
+        try {
+            @SuppressWarnings("JavaReflectionMemberAccess")
+            final Field field = AbstractMeterRegistry.class.getDeclaredField("meterMap");
```

Comment:
If I understand correctly, I think the JVM can optimize reflection much better, including inlining, if the field and method handles are class constants.